### PR TITLE
 PUT for Driver Availability Controller (Driver Availability) 

### LINKED ISSUE #12: FEATURE: PUT for Driver Availability Controller (Driver Availability)
Acceptance Criteria: 

- [ ] `PUT /api/driverAvailability` that edits an availability if owned by current user.

### PR DESCRIPTION
**Pull request by notaryaman**
adds PUT endpoint for Driver Availability Controller, enter ID for in ID space to delete availability

working deployment: https://project-notaryaman-dev.dokku-16.cs.ucsb.edu/

image

image

closes #12

### CODE REVIEW

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]
Inadequate PR description

https://ucsb-cs156.github.io/topics/pull_requests/

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 1]
```diff
@@ -64,5 +64,40 @@ public DriverAvailability postDriverAvailability(
         DriverAvailability savedDriverAvailability = driverAvailabilityRepository.save(driverAvailability);
 
         return savedDriverAvailability;
+        }
+
+    @Operation(summary= "Delete Driver Availability")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @DeleteMapping("")
+    public Object deletemenuitem(
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:72]: Drivers should only be able to delete their own availabilities, and admins should be able to delete anyones

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 2]
```diff
@@ -64,5 +64,40 @@ public DriverAvailability postDriverAvailability(
...
+        
+         driverAvailabilityRepository.delete(drivav);
+        return genericMessage("Driver Availability with id %s deleted".formatted(id));
+        }
+    
+
+    @Operation(summary= "Update Driver Availability")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @PutMapping("")
+    public DriverAvailability updatemenuitem(
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:86]: Drivers should only be able to update their own, admins should be able to edit anything

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]
Also please give admins the ability to create availabilities

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 3]
```diff
@@ -77,7 +77,49 @@ public Object deletemenuitem(
...
+    @GetMapping("/all")
+    public Iterable<DriverAvailability> allmenuitems() {
+        Iterable<DriverAvailability> drivall = driverAvailabilityRepository.findAll();
+        return drivall;
+    }
+
+
+    @Operation(summary= "Get Driver Availability")
+    @PreAuthorize("hasAnyRole('ADMIN', 'DRIVER','USER')")
+    @GetMapping("")
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:115]: Regular users should not be able to view driver availabilities, could be pontentially harmful

Other than that good to merge, ping me after change

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java (hunk 1 above)
Comment (andrewpengucsb) [reply src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:72]: Drivers are still able to whatever availability, you need to check which role the user has inside the method and based on that determine whether they can delete or not

Please see existing controller https://github.com/ucsb-cs156/proj-gauchoride/blob/main/src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java (hunk 2 above)
Comment (andrewpengucsb) [reply src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:85]: Same here as delete

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java [hunk 4]
```diff
@@ -77,7 +77,49 @@ public Object deletemenuitem(
...
+        return drivav;
+    }
+
+
+    @Operation(summary= "List all Driver Availability")
+    @PreAuthorize("hasAnyRole('ADMIN', 'DRIVER','USER')")
+    @GetMapping("/all")
+    public Iterable<DriverAvailability> allmenuitems() {
+        Iterable<DriverAvailability> drivall = driverAvailabilityRepository.findAll();
+        return drivall;
```
Comment (andrewpengucsb) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java:109]: Same comment as the get single

**Review by andrewpengucsb — CHANGES_REQUESTED** [review]

**Review by andrewpengucsb — APPROVED** [review]
LGTM

**Review by AllenHsm — APPROVED** [review]
LGTM! Nice job

**Pull request merged**